src/util.js

```diff
@@ -1,36 +1,18 @@
 import {ChartInternal} from './chart-internal';
+import {Chart} from './chart';
```

Comment:
Not sure if this is the scope of this PR but in my opinion the util.js file should NOT import chart/chart internal.

This file should be as "low level" at it can be.

Otherwise we are bound to have circular dependency when importing this file

Reply:
You mean `util.js` should be low level in the sense of being just a collection of functions? In that case, `c3` could be elsewhere.

However, `util.js` exports `Component` which in turn is used by `Axis` to extend `ChartInternal` and `c3.chart.internal`. This means `Component` needs the definition of  `ChartInternal` and  `c3` (of course) and you can't define `c3` without importing `Chart` and `ChartInternal`.

However, I'm confused, because `Axis` is instanced as

```js
$$.axis = new Axis($$)
```
(where $$ is the instance of ChartInternal).

And when instancing Axis, the side effect are:

1.- `Axis (instance).owner = ChartInternal instance`

2.- `c3` is changed from:

```js
var c3 = {
    version: "0.6.3",
    chart: {
        fn: Chart.prototype,
        internal: {
            fn: ChartInternal.prototype,
        }
    },
    generate: function(config) {
        return new Chart(config);
    }
};
```

to 

```js
var c3 = {
    version: "0.6.3",
    chart: {
        fn: Chart.prototype,
        internal: {
            fn: ChartInternal.prototype,
            axis: {
                fn: Axis.prototype,
                internal: {
                    fn: AxisInternal.prototype
                }
            }
        }
    },
    generate: function(config) {
        return new Chart(config);
    }
};

```

the last one could be handled if we moved the definition of `c3` to a separate file which imported `Chart`, `ChartInternal`, `Axis` and `AxisInternal`.

whereas the first one could be handled simply by defining Axis as

```js
export default class Axis {
    constructor(owner) {
        this.owner = owner;
        this.d3 = owner.d3;
        this.internal = AxisInternal;
    }
}
```

`util.js` also exports functions `isEmpty` and `notEmpty` where the second one is defined as

```js
export var notEmpty = function (o) {
    return !ChartInternal.prototype.isEmpty(o);
};
```

again, this would mean that `util.js` needs `ChartInternal`. BUT, it would be as simple as declaring `notEmpty` as

```js
export var notEmpty = function (o) {
    return !isEmpty(o);
};
```

Because it is by all means a static function, even if they are later attached to `ChartInternal.prototype`.

I'll make some arrangements to see if, with some refactoring, this could pass the tests keeping `util.js` just a bunch of static functions.

Reply:
Sorry for such a short response to such a huge reply but:

> You mean util.js should be low level in the sense of being just a collection of functions? In that case, c3 could be elsewhere.

That's exactly what I mean !

We shouldn't have to depend on the whole "ChartInternal" class for such small functions.

> However, util.js exports Component which in turn is used by Axis to extend ChartInternal and c3.chart.internal. This means Component needs the definition of ChartInternal and c3 (of course) and you can't define c3 without importing Chart and ChartInternal.

And that is why :)

> Because it is by all means a static function, even if they are later attached to 

I think they shouldn't even be attached to the prototype... They are just static utility functions that you import when needed.